06_Userspace/06_Resource_Management.md

```diff
@@ -0,0 +1,118 @@
+# Resource Management
+
+The kernel manages various things on behalf of the userspace process, including files, sockets, IPC, devices, and more, which we call _resources_. It is a good idea to have a unified way to handle such a range of resources to reduce complexity. Rather than exposing a separate set of syscalls for each resource, a generic abstractation can be introduced to simplify everything while also keeping it all centralized in one place. 
+
+To do this, we implement an API where every resource can be opened, read from, written to, and closed using the same syscalls. Through this design, the kernel is kept small, whilst also letting new resources be added in the future with minimal change to both kernel and user code.
+
+## Resource Abstractation
+
+When talking about _resources_, we need a way to distinguish between the different types that the kernel may expect to provide to userspace. Each resource behaves differently internally, but from the view of the userspace process, everything should be accessible from the same set of syscalls. In order to achieve this, we define an enum of resource types to allow the kernel to tag each resource with its category. This way, when a system call is made, the kernel knows how to dispatch the request.
```

Comment:
Even if in  the example is an enum, I don't think that it has to be strictly it. 
The sentence is kind of assuming and telling it, I think that it can be made more generic. 

Something like: ` In order to achieve this we need to define a set of resource types to allow the kernel to tag each resources, for example we can use an enum... etc.`